sources/common/src/LogParser.cpp

```diff
@@ -44,3 +44,26 @@ PARSER::LogParser::LogParser(const std::string &filename,
                                  "' failed, it either doesn't exist or is inaccessible.");
     }
 }
+
+void PARSER::LogParser::Attach(INSTRUMENT::Instrument &instrument)
+{
+    m_vpInstruments.push_back(&instrument);
+}
+
+void PARSER::LogParser::TerminateInstruments()
+{
+    for (auto instrument : m_vpInstruments)
+    {
+        instrument->Terminate();
+    }
+}
+
+void PARSER::LogParser::Execute()
+{
+    // Process the log.
+    Parse();
+
+    // Trigger the final action of instruments.
+    TerminateInstruments();
+}
```

Comment:
Comments above are useless. If you feel need for a comment in such place, change function name to better one.
Could be inlined in hpp.